addons/l10n_account_withholding/models/account_tax.py

```diff
@@ -0,0 +1,229 @@
+    The withholding line amounts are the same as for price-excluded taxes. (this will be the case for each scenario)
+    This means that when calculating the withholding line amount for price-included withholding tax,
+    the calculation should treat these as tax-excluded taxes.
+
+    Scenario 7:
+    ‾‾‾‾‾‾‾‾‾‾‾
+    Product Line                                  Withholding Line                  Payment Total
+    ______________________________________       __________________________                  1035
+   | Price Unit | Tax  | SubTotal | Total |     | Base Amount | Tax Amount |
+   |       1000 | C, B |     1000 |  1150 |     |        1150 |        115 |
+    ¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯      ¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯
+
+    Scenario 8:
+    ‾‾‾‾‾‾‾‾‾‾‾
+    Product Line                                  Withholding Line                  Payment Total
+    ______________________________________       __________________________                   900
+   | Price Unit | Tax  | SubTotal | Total |     | Base Amount | Tax Amount |
+   |       1000 | E, B |   869.57 |  1000 |     |        1000 |        100 |
+    ¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯      ¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯
+
+    Scenario 9:
+    ‾‾‾‾‾‾‾‾‾‾‾
+    Product Line                                  Withholding Line                  Payment Total
+    ______________________________________       __________________________                  1050
+   | Price Unit | Tax  | SubTotal | Total |     | Base Amount | Tax Amount |
+   |       1000 | D, B |    1000  |  1150 |     |        1000 |        100 |
+    ¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯      ¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯
+
+    Scenario 10:
+    ‾‾‾‾‾‾‾‾‾‾‾
+    Product Line                                  Withholding Line                  Payment Total
+    ______________________________________       __________________________                913.04
+   | Price Unit | Tax  | SubTotal | Total |     | Base Amount | Tax Amount |
+   |       1000 | F, B |  869.57  |  1000 |     |      869.57 |      86.96 |
+    ¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯      ¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯
+
+    Note that these cases do not include multiple withholding taxes; the same logic would apply.
+    Withholding taxes can affect each other, in which case it will be visible in the payment register.
+    """
+    _inherit = 'account.tax'
+
+    # ------------------
+    # Fields declaration
+    # ------------------
+
+    is_withholding_tax_on_payment = fields.Boolean(
+        string="Withholding On Payment",
+        help="If enabled, this tax will not affect journal entries until the registration of payment.",
+    )
+    withholding_sequence_id = fields.Many2one(
+        string='Withholding Sequence',
+        help='Label displayed on Journal Items and Payment Receipts.',
+        comodel_name='ir.sequence',
+        copy=False,
+        check_company=True,
+    )
+    color = fields.Integer(compute='_compute_color')
```

Comment:
I think the field/compute should be in the account module

Reply:
I'll reply to a few comments today, we can check the rest Monday :)

yes this would make more sense; but this field isn't really in the spec.
I added it as it makes it easier to distinguish the taxes while testing, but it wasn't really talked about with Benj nor approved.